src/main.ts

```diff
@@ -86,13 +87,16 @@ export async function run(): Promise<void> {
   } catch (err) {
     // Fail the workflow run if an error occurs
     core.setFailed(
-      err instanceof Error ? err.message : /* istanbul ignore next */ `${err}`
+      err instanceof Error ? err : /* istanbul ignore next */ `${err}`
```

Comment:
Passing the entire `Error` object causes the error class to be displayed in the output.